Core/Src/Sensors/AnalogSensor.c

```diff
@@ -1,7 +1,187 @@
 #include "../../Inc/Sensors/AnalogSensor.h"
+#include <stdio.h>
+
+void initAnalogSensor(AnalogSensor* sensor, const char* name, int hz, int channel) {
+    initSensor(&sensor->base, name, hz);
+    sensor->channel = channel;
+
+    if(sensor->channel == 0) {
+    	// HAL code to bind to PA0
+    	__HAL_RCC_GPIOA_CLK_ENABLE();
+    	GPIO_InitTypeDef GPIOAout_Struct;
+		GPIOJout_Struct.Pin = GPIO_PIN_0;
+		GPIOJout_Struct.Mode = GPIO_MODE_OUTPUT_PP;
```

Comment:
Since this statement and other statements are in every case you could move them out of the conditionals.